File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    int same = 0;
    int zero = 0;

    for (int lotto : lottos) {
        if (find(win_nums.begin(), win_nums.end(), lotto) != win_nums.end()) {
            same++;
        }
        if (lotto == 0) {
            zero++;
        }
    }

    int best = (same + zero == 0) ? 6 : 7 - (same + zero);
    int worst = (same == 0) ? 6 : 7 - same;

    return {best, worst};
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
from typing import List

def _rank(matches: int) -> int:
    mapping = {6: 1, 5: 2, 4: 3, 3: 4, 2: 5}
    return mapping.get(matches, 6)

def solution(lottos: List[int], win_nums: List[int]) -> List[int]:
    zero_count = lottos.count(0)
    matched = len(set(lottos) & set(win_nums))
    best = _rank(matched + zero_count)
    worst = _rank(matched)
    return [best, worst]
